### ISSUE TITLE
Get wp-publications to update when dependent @string updates

### ISSUE THREAD

**Issue by davidlie** [monperrus/wp-publications#1]
I was wrestling with an issue where I had updated a @string but wp-publications seems to only update when the actual bibentry updates.  For example if I have a bibentry:

```
@string{asiaccs2022="Proceedings of the 17th ACM ASIA Conference on Computer and Communications Security (ASIACCS)"}
@inproceedings{foo, 
	title = {Baz},
	author = {Alice and Bob},
	booktitle = ASIACCS2022,
	month = may,
	year = 2022,
}
```
If I only update the @string and not the @inproceedings, then the display for the booktitle won't update.  I've been looking at the code and trying to figure out how to fix this but don't really see how it is checking whether an entry has changed, even though it is somehow realizing that the entry does depend on the @string.  It will only update if I directly modify one of the fields in @inproceedings.  Any pointers?

**Comment by monperrus** [comment]
Thanks for the bug report. In theory, this should work because we cache based on the full file, not based on specific entries: https://github.com/monperrus/wp-publications/blob/master/bibtexbrowser.php#L397

**Comment by davidlie** [comment]
Thanks for the response. I think the issue might be at https://github.com/monperrus/wp-publications/blob/3246a6df5c7e3f5d34e932368da4ee0503dfb7d1/bibtexbrowser.php#L3727, as if I update a string without changing the key, the default behavior of `array_merge`, if not I'm not mistaken, is to keep the first value, which is the old one cached in the compiledbib.  I haven't tested it, but reversing the order of `$this->stringdb` and `$db->stringdb` should fix the problem?

**Comment by monperrus** [comment]
Thanks for the analysis. Would you be able to make a pull-request?

**Comment by davidlie** [comment]
Yes, I am running on an older version so I would need to update to the latest one and then submit the PR.

**Comment by monperrus** [comment]
ack, looking forward to it!

**Comment by davidlie** [comment]
Sorry, unrelated but while porting my page to the latest version, I've run into a problem.  I have shortcode:
`[wp-publications bib=XXX key=XXX_list layout=ordered_list]` and this used to generate a list without year separators.  Now I can't seem to make it generate the same list without the year separators.  Do I need to pass some option to suppress the year separator generation?

**Comment by monperrus** [comment]
Would you open a new issue about shortcodes? Thanks!

**Comment by davidlie** [comment]
Actually, I can't remember how I fixed the shortcode issue.  I think might have been a customizations I've made to the code base that was causing the issue, nothing in the base issue.  I did find a problem with my original fix though, which is that it wouldn't update on cases where changes to the bibtex entry wouldn't affect the listing, but might affect the bibtex post a user clicks to see the underlying bibtex. Thus, it's better to compare the unformatted bibtex entry, which has the string substitutions done, but also includes fields that might not be displayed in the listing.
